### PR TITLE
Define cpp standard only if not defined

### PR DESCRIPTION
**Pull request by MartinDelille**
I'm working on the following conan package for QXlsx and according to [this comment](https://github.com/conan-io/conan-center-index/pull/11364#issuecomment-1178477671), it would be nice to respect `CMAKE_CXX_STANDARD` if it is already defined.

### CODE REVIEW

**Comment by paulharris** [comment]
I took it to the wise people of cmake-slack,
suggestions:
* set min cmake to 3.21
* don't do the if (NOT CMAKE_CXX_STANDARD) check, instead just set as a cache variable, ie
set (CMAKE_CXX_STANDARD 17 CACHE STRING "")
* change add_definitions() to target_compile_definitions()
* add a: target_compile_features() to enforce a minimum C++ standard version

https://cpplang.slack.com/archives/C5Y2GDECX/p1658498760165389

**Comment by dantti** [comment]
I don't see why cmake 3.21 should be required, nor why force cpp 17 on Qt5

File: QXlsx/CMakeLists.txt [hunk 1]
```diff
@@ -152,6 +139,22 @@ add_library(QXlsx
...
+    -DQT_STRICT_ITERATORS
+    -DQT_NO_URL_CAST_FROM_STRING
+    -DQT_NO_CAST_FROM_BYTEARRAY
+    -DQT_USE_QSTRINGBUILDER
+    -DQT_NO_SIGNALS_SLOTS_KEYWORDS
+    -DQT_USE_FAST_OPERATOR_PLUS
+    -DQT_DISABLE_DEPRECATED_BEFORE=0x050F00
+)
+
+target_compile_features(QXlsx INTERFACE cxx17)
```
Comment (dantti) [inline QXlsx/CMakeLists.txt:156]: Afaik there is no cxx17 usage, so cxx_std_11 would make more sense

Comment (MartinDelille) [reply QXlsx/CMakeLists.txt:156]: Indeed.

**Comment by MartinDelille** [comment]
Tell me if you want me to squash the commits.

**Comment by MartinDelille** [comment]
68abfe5 => I just watch this and the diff hurted me: https://www.youtube.com/watch?v=SsoOG6ZeyUI

**Comment by dantti** [comment]
squashed and merged thanks

**Comment by paulharris** [comment]
> [68abfe5](https://github.com/QtExcel/QXlsx/commit/68abfe5a5523e6717734e4db80e57623cba2e4ea) => I just watch this and the diff hurted me: https://www.youtube.com/watch?v=SsoOG6ZeyUI

what... who uses vim and actually presses the spacebar instead of the tab... `set expandtab` for a start ... the girl is doing it wrong!

**Comment by MartinDelille** [comment]
@dantti I already asked you but do you plan to release a new version including the latest Qt 6 compatibility fix ?

**Comment by dantti** [comment]
@MartinDelille I think I can do one now

**Comment by MartinDelille** [comment]
Any idea when the release will happen ? 😄

**Comment by dantti** [comment]
Sorry, done now...

**Comment by MartinDelille** [comment]
Thank you so much! 🎉

**Pull request merged**